src/plugins/intel_cpu/src/nodes/eye.cpp

```diff
@@ -29,8 +30,26 @@ bool Eye::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::s
     return true;
 }
 
+namespace {
+class EyeShapeInferFactory : public ShapeInferFactory {
+public:
+    EyeShapeInferFactory(std::shared_ptr<ov::Node> op) : m_op(op) {}
+    ShapeInferPtr makeShapeInfer() const override {
+        IShapeInfer::port_mask_t port_mask = EMPTY_PORT_MASK;
+        if (m_op->get_input_size() == 4) {
```

Comment:
BTW thats interesting story. Should we always care about optional inputs on each particular factory level? Or developer can always set all port dependencies (even for optional inputs) while common logic will handle only really existing ones?

Reply:
Ok. Lets do in follow-up PR.